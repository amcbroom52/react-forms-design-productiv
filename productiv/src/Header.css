.Header-quote-area {
  text-align: right;
}

.Header-button {
  color: white;
  background-color: rgb(145, 107, 10);
}