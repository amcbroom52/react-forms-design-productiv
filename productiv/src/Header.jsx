import "./Header.css";

/** Renders page header
 *
 * Props:
 * - quote
 * - generateQuote()
 *
 * State:
 * - none
 *
 * HeaderApp -> Header
 *
 */

function Header({ quote = "", generateQuote }) {

  return (
    <header className="container-fluid pt-4 pb-1">
      <div className="container">
        <h1>Prøductïv</h1>
        <p className="lead">The best name in todo list management.</p>
      </div>
      <div className="container Header-quote-area">
        <p className="Header-quote">{ quote && `${quote.text} -${quote.author}`}</p>
        <button
          onClick={generateQuote}
          className="Header-button">
          {quote
            ? "Nü quøte"
            : "Click here for an inspirational quøte!"}
        </button>
      </div>
    </header>
  );
}

export default Header;