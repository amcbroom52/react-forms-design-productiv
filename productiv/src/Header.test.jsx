import { describe, it, expect } from "vitest";

import React from "react";
import { render } from "@testing-library/react";
import Header from "./Header";

describe("Header component", function () {
  it("renders without crashing", function () {
    render(<Header />);
  });

  it("matches snapshot without quote", function () {
    const { asFragment } = render(<Header />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("matches snapshot with quote", function () {
    const { asFragment } = render(
      <Header quote={{ text: "randomness", author: "Adrian McBroom" }} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("contains Quote Text", function () {
    const { container } = render(
      <Header quote={{ text: "randomness", author: "Adrian McBroom" }} />
    );

    expect(container).toContainHTML("randomness -Adrian McBroom");
  });
});

