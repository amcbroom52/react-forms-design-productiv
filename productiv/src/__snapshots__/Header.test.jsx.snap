// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header component > matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="container-fluid pt-4 pb-1"
  >
    <div
      class="container"
    >
      <h1>
        Prøductïv
      </h1>
      <p
        class="lead"
      >
        The best name in todo list management.
      </p>
    </div>
    <div
      class="container Header-quote-area"
    >
      <p
        class="Header-quote"
      />
      <button
        class="Header-button"
      >
        Click here for an inspirational quøte!
      </button>
    </div>
  </header>
</DocumentFragment>
`;

exports[`Header component > matches snapshot 2`] = `
<DocumentFragment>
  <header
    class="container-fluid pt-4 pb-1"
  >
    <div
      class="container"
    >
      <h1>
        Prøductïv
      </h1>
      <p
        class="lead"
      >
        The best name in todo list management.
      </p>
    </div>
    <div
      class="container Header-quote-area"
    >
      <p
        class="Header-quote"
      />
      <button
        class="Header-button"
      >
        Click here for an inspirational quøte!
      </button>
    </div>
  </header>
</DocumentFragment>
`;

exports[`Header component > matches snapshot with quote 1`] = `
<DocumentFragment>
  <header
    class="container-fluid pt-4 pb-1"
  >
    <div
      class="container"
    >
      <h1>
        Prøductïv
      </h1>
      <p
        class="lead"
      >
        The best name in todo list management.
      </p>
    </div>
    <div
      class="container Header-quote-area"
    >
      <p
        class="Header-quote"
      >
        randomness -Adrian McBroom
      </p>
      <button
        class="Header-button"
      >
        Nü quøte
      </button>
    </div>
  </header>
</DocumentFragment>
`;

exports[`Header component > matches snapshot without quote 1`] = `
<DocumentFragment>
  <header
    class="container-fluid pt-4 pb-1"
  >
    <div
      class="container"
    >
      <h1>
        Prøductïv
      </h1>
      <p
        class="lead"
      >
        The best name in todo list management.
      </p>
    </div>
    <div
      class="container Header-quote-area"
    >
      <p
        class="Header-quote"
      />
      <button
        class="Header-button"
      >
        Click here for an inspirational quøte!
      </button>
    </div>
  </header>
</DocumentFragment>
`;
